aiohttp/client_reqrep.py

```diff
@@ -433,7 +433,7 @@ def update_auth(self, auth: Optional[BasicAuth]) -> None:
         self.headers[hdrs.AUTHORIZATION] = auth.encode()
 
     def update_body_from_data(self, body: Any) -> None:
-        if not body:
+        if body is None:
```

Comment:
This one fixes the error reported. If somebody more familiar with the code can double check that the other 2 are sensible changes, that would be great.